### PR TITLE
[NFC] Merge `test/type/explicit_existential` & `test/type/explicit_existential_swift6`

### PR DESCRIPTION
**Pull request by AnthonyLatsis**

### CODE REVIEW

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

File: test/type/explicit_existential.swift [hunk 1]
```diff
@@ -1,4 +1,5 @@
-// RUN: %target-typecheck-verify-swift -enable-experimental-feature NoncopyableGenerics
+// RUN: %target-typecheck-verify-swift
+// RUN: %target-typecheck-verify-swift -enable-upcoming-feature ExistentialAny -verify-additional-prefix explicit-any-
```
Comment (xedin) [inline test/type/explicit_existential.swift:2]: I'm not sure why `swift6` version did not have it but please add one more run command with `-swift-version 6` for completeness.

Comment (AnthonyLatsis) [reply test/type/explicit_existential.swift:2]: Good idea. Note that we have excluded this feature from `-swift-version 6`.

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Review by xedin — APPROVED** [review]

**Pull request merged**